src/turbomind/models/llama/LlamaBatch.cc

```diff
@@ -722,8 +722,8 @@ void LlamaBatch<T>::AllocateBuffer(size_t batch_size, size_t session_len, int ca
             context_decoder_output_buf_, sizeof(T) * max_forward_token_num_ * hidden_units, false);
     }
 
-    context_decoder_input_buf_ =
-        (T*)allocator_->reMalloc(context_decoder_input_buf_, sizeof(T) * max_forward_token_num_ * hidden_units, false);
+    context_decoder_input_buf_ = (T*)allocator_->reMalloc(
+        context_decoder_input_buf_, sizeof(T) * max_forward_token_num_ * hidden_units * 2, false);
```

Comment:
why *2?

Reply:
Split embedding weights along hidden dim needs additional buffer with shape TP x num_token x {hidden / TP}.

After embedding layer, the result buffer needs to permute with dims (1, 0, 2)

Reply:
Re-purpose `context_decoder_output_buf_ ` instead of doubling the input buffer.